packages/slice-machine/lib/models/common/http/DefaultClient.ts

```diff
@@ -129,21 +130,23 @@ export default class DefaultClient {
     auth: string
   ): Promise<Error | UserProfile> {
     try {
-      const result = await fetch(`${createApiUrl(base, UserService)}profile`, {
-        method: "GET",
-        headers: {
-          Authorization: `Bearer ${auth}`,
-        },
-      });
+      const result = await axios.get<UserProfile>(
+        `${createApiUrl(base, UserService)}profile`,
+        {
+          headers: {
+            Authorization: `Bearer ${auth}`,
+          },
+        }
+      );
 
       // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const jsResult = await result.json();
 
       return getOrElseW(
         // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-        () => new Error(`Unable to parse profile: ${jsResult}`)
-      )(UserProfile.decode(jsResult));
+        () => new Error(`Unable to parse profile: ${result.data}`)
+      )(UserProfile.decode(result.data));
     } catch (e) {
+      // TODO: why not throw?
       return e as Error;
```

Comment:
I think we have to discuss on what is our Error management strategy, for now we had 2 strategies in the code : 
- Create Error object in the lower-level functions and add some condition on the top-level functions
- Throw error on lower-level functions and try catch on top-level functions

Reply:
yeah the return type here doesn't make much sense. `Promise<UserProfile | Error>` means that the response had to be cast later on

Reply:
I would throw, 
by the looks of this we'd need to have some kind of pattern matching to figure out there was an error, which at that point we may as well be using scala.js :/